### ISSUE TITLE
Doctests failures in sage/doctest/control.py when sagemath-meataxe is installed

### ISSUE THREAD

**Issue by kiwifb** [sagemath/sage#37905]
### Steps To Reproduce

This is observed in sage-on-gentoo with both python 3.11 and python 3.12 and sage 10.4.beta4. Install sagemath-meataxe and run 
```
sage -t --long /usr/lib/python3.11/site-packages/sage/doctest/control.py
```

### Expected Behavior

When sagemath-meataxe is not installed, the doctest pass
```
sage -t --long --random-seed=309974202432976880116456820555380509718 /usr/lib/python3.11/site-packages/sage/doctest/control.py
    [229 tests, 1.46 s]
----------------------------------------------------------------------
All tests passed!
----------------------------------------------------------------------
```

### Actual Behavior

The doctest fails with
```
sage -t --long --random-seed=170181526529337143612090850685532611974 /usr/lib/python3.11/site-packages/sage/doctest/control.py
**********************************************************************
File "/usr/lib/python3.11/site-packages/sage/doctest/control.py", line 1447, in sage.doctest.control.DocTestController.run
Failed example:
    DC.run()
Expected:
    Running doctests with ID ...
    Using --optional=sage...
    Features to be detected: ...
    Doctesting 1 file.
    sage -t ....py
        [4 tests, ... s]
    ----------------------------------------------------------------------
    All tests passed!
    ----------------------------------------------------------------------
    Total time for all tests: ... seconds
        cpu time: ... seconds
        cumulative wall time: ... seconds
    Features detected...
    0
Got:
    Running doctests with ID 2024-04-30-13-53-30-5803c5d0.
    Running with SAGE_LOCAL='/usr' and SAGE_VENV='/usr'
    Using --optional=sage
    Features to be detected: 4ti2,benzene,bliss,buckygen,conway_polynomials,csdp,cvxopt,cvxopt,database_cremona_ellcurve,database_cremona_mini_ellcurve,database_cubic_hecke,database_ellcurves,database_graphs,database_jones_numfield,database_knotinfo,dvipng,ecm,fpylll,fricas,gap_package_atlasrep,gap_package_design,gap_package_grape,gap_package_guava,gap_package_hap,gap_package_polycyclic,gap_package_qpa,gap_package_quagroup,gfan,graphviz,imagemagick,ipython,jmol,jupymake,kenzo,latte_int,lrcalc_python,lrslib,matroid_database,mcqd,meataxe,mpmath,msolve,nauty,networkx,numpy,palp,pandoc,pdf2svg,pdftocairo,pexpect,phitigra,pillow,plantri,polytopes_db,polytopes_db_4d,pplpy,primecountpy,ptyprocess,pynormaliz,pyparsing,python_igraph,requests,rubiks,sage.combinat,sage.geometry.polyhedron,sage.graphs,sage.groups,sage.libs.braiding,sage.libs.ecl,sage.libs.flint,sage.libs.gap,sage.libs.linbox,sage.libs.m4ri,sage.libs.ntl,sage.libs.pari,sage.libs.singular,sage.misc.cython,sage.modular,sage.modules,sage.numerical.mip,sage.plot,sage.rings.complex_double,sage.rings.finite_rings,sage.rings.function_field,sage.rings.number_field,sage.rings.padics,sage.rings.polynomial.pbori,sage.rings.real_double,sage.rings.real_mpfr,sage.sat,sage.schemes,sage.symbolic,sage_numerical_backends_coin,sagemath_doc_html,scipy,singular,sphinx,symengine_py,sympy,tdlib,threejs
    Doctesting 1 file.
    sage -t --random-seed=0 /tmp/tmp3o72lmtj/tmp_qd1plq6l.py
        [5 tests, 0.12 s]
    ----------------------------------------------------------------------
    All tests passed!
    ----------------------------------------------------------------------
    Total time for all tests: 0.1 seconds
        cpu time: 0.1 seconds
        cumulative wall time: 0.1 seconds
    Features detected for doctesting: meataxe,sage.libs.singular,sage.modules,sage.rings.finite_rings,sage.rings.number_field,sage.rings.real_mpfr,sage.symbolic,sympy
    Features that have been hidden: meataxe
    0
**********************************************************************
File "/usr/lib/python3.11/site-packages/sage/doctest/control.py", line 1465, in sage.doctest.control.DocTestController.run
Failed example:
    DC.run()
Expected:
    Running doctests with ID ...
    Using --optional=sage
    Features to be detected: ...
    Doctesting 1 file.
    sage -t ....py
        [4 tests, ... s]
    ----------------------------------------------------------------------
    All tests passed!
    ----------------------------------------------------------------------
    Total time for all tests: ... seconds
        cpu time: ... seconds
        cumulative wall time: ... seconds
    Features detected...
    0
Got:
    Running doctests with ID 2024-04-30-13-53-30-601ce747.
    Running with SAGE_LOCAL='/usr' and SAGE_VENV='/usr'
    Using --optional=sage
    Features to be detected: 4ti2,bliss,conway_polynomials,csdp,cvxopt,cvxopt,database_cremona_ellcurve,database_cremona_mini_ellcurve,database_cubic_hecke,database_ellcurves,database_graphs,database_jones_numfield,database_knotinfo,dvipng,ecm,fpylll,fricas,gap_package_atlasrep,gap_package_design,gap_package_grape,gap_package_guava,gap_package_hap,gap_package_polycyclic,gap_package_qpa,gap_package_quagroup,gfan,graphviz,imagemagick,ipython,jmol,jupymake,kenzo,latte_int,lrcalc_python,lrslib,matroid_database,mcqd,meataxe,mpmath,msolve,nauty,networkx,numpy,palp,pandoc,pdf2svg,pdftocairo,pexpect,phitigra,pillow,plantri,polytopes_db,polytopes_db_4d,pplpy,primecountpy,ptyprocess,pynormaliz,pyparsing,python_igraph,requests,rubiks,sage.combinat,sage.geometry.polyhedron,sage.graphs,sage.groups,sage.libs.braiding,sage.libs.ecl,sage.libs.flint,sage.libs.gap,sage.libs.linbox,sage.libs.m4ri,sage.libs.ntl,sage.libs.pari,sage.libs.singular,sage.misc.cython,sage.modular,sage.modules,sage.numerical.mip,sage.plot,sage.rings.complex_double,sage.rings.finite_rings,sage.rings.function_field,sage.rings.number_field,sage.rings.padics,sage.rings.polynomial.pbori,sage.rings.real_double,sage.rings.real_mpfr,sage.sat,sage.schemes,sage.symbolic,sage_numerical_backends_coin,sagemath_doc_html,scipy,singular,sphinx,symengine_py,sympy,tdlib,threejs
    Doctesting 1 file.
    sage -t --random-seed=0 /tmp/tmp3o72lmtj/tmp_qd1plq6l.py
        [5 tests, 0.14 s]
    ----------------------------------------------------------------------
    All tests passed!
    ----------------------------------------------------------------------
    Total time for all tests: 0.2 seconds
        cpu time: 0.1 seconds
        cumulative wall time: 0.1 seconds
    Features detected for doctesting: meataxe,sage.libs.singular,sage.modules,sage.rings.finite_rings,sage.rings.number_field,sage.rings.real_mpfr,sage.symbolic,sympy
    Features that have been hidden: meataxe
    0
**********************************************************************
File "/usr/lib/python3.11/site-packages/sage/doctest/control.py", line 1483, in sage.doctest.control.DocTestController.run
Failed example:
    DC.run()                              # optional - meataxe
Expected:
    Running doctests with ID ...
    Using --optional=sage
    Features to be detected: ...
    Doctesting 1 file.
    sage -t ....py
        [4 tests, ... s]
    ----------------------------------------------------------------------
    All tests passed!
    ----------------------------------------------------------------------
    Total time for all tests: ... seconds
        cpu time: ... seconds
        cumulative wall time: ... seconds
    Features detected...
    Features that have been hidden: ...meataxe...
    0
Got:
    Running doctests with ID 2024-04-30-13-53-30-a0fa6d56.
    Running with SAGE_LOCAL='/usr' and SAGE_VENV='/usr'
    Using --optional=sage
    Features to be detected: 4ti2,bliss,conway_polynomials,csdp,cvxopt,cvxopt,database_cremona_ellcurve,database_cremona_mini_ellcurve,database_cubic_hecke,database_ellcurves,database_graphs,database_jones_numfield,database_knotinfo,dvipng,ecm,fpylll,fricas,gap_package_atlasrep,gap_package_design,gap_package_grape,gap_package_guava,gap_package_hap,gap_package_polycyclic,gap_package_qpa,gap_package_quagroup,gfan,graphviz,imagemagick,ipython,jmol,jupymake,kenzo,latte_int,lrcalc_python,lrslib,matroid_database,mcqd,meataxe,mpmath,msolve,nauty,networkx,numpy,palp,pandoc,pdf2svg,pdftocairo,pexpect,phitigra,pillow,plantri,polytopes_db,polytopes_db_4d,pplpy,primecountpy,ptyprocess,pynormaliz,pyparsing,python_igraph,requests,rubiks,sage.combinat,sage.geometry.polyhedron,sage.graphs,sage.groups,sage.libs.braiding,sage.libs.ecl,sage.libs.flint,sage.libs.gap,sage.libs.linbox,sage.libs.m4ri,sage.libs.ntl,sage.libs.pari,sage.libs.singular,sage.misc.cython,sage.modular,sage.modules,sage.numerical.mip,sage.plot,sage.rings.complex_double,sage.rings.finite_rings,sage.rings.function_field,sage.rings.number_field,sage.rings.padics,sage.rings.polynomial.pbori,sage.rings.real_double,sage.rings.real_mpfr,sage.sat,sage.schemes,sage.symbolic,sage_numerical_backends_coin,sagemath_doc_html,scipy,singular,sphinx,symengine_py,sympy,tdlib,threejs
    Doctesting 1 file.
    sage -t --random-seed=0 /tmp/tmp3o72lmtj/tmp_qd1plq6l.py
        [5 tests, 0.12 s]
    ----------------------------------------------------------------------
    All tests passed!
    ----------------------------------------------------------------------
    Total time for all tests: 0.1 seconds
        cpu time: 0.1 seconds
        cumulative wall time: 0.1 seconds
    Features detected for doctesting: meataxe,sage.libs.singular,sage.modules,sage.rings.finite_rings,sage.rings.number_field,sage.rings.real_mpfr,sage.symbolic,sympy
    Features that have been hidden: meataxe
    0
**********************************************************************
1 item had failures:
   3 of  23 in sage.doctest.control.DocTestController.run
    [230 tests, 3 failures, 1.64 s]
----------------------------------------------------------------------
sage -t --long --random-seed=170181526529337143612090850685532611974 /usr/lib/python3.11/site-packages/sage/doctest/control.py  # 3 doctests failed
```

### Additional Information

As can be seen in the failing doctest there is a mention that meataxe is a hidden feature. But in actual fact it is not during this test.

### Environment

```markdown
- Gentoo
- 10.4.beta4 (sage-on-gentoo)
```

### Checklist

- [X] I have searched the existing issues for a bug report that matches the one I want to file, without success.
- [X] I have read the documentation and troubleshoot guide

**Comment by kiwifb** [comment]
Actually it is most likely a self inflicted sage-on-gentoo feature because I avoid using stuff from misc/package.py in doctest as much as possible. But if we are going to deal with that unfortunate old sage feature we need to remember about this.

This test failure also only appeared recently.

**Comment by kiwifb** [comment]
On closer inspection, I am not sure how meataxe is supposed to be masked in that doctest. I cannot work it out.

**Comment by mkoeppe** [comment]
@kiwifb See #37857 for some of the necessary changes; but also there is an open question.

**Comment by soehms** [comment]
The issue will be fixed by #37737, especially commit https://github.com/sagemath/sage/pull/37737/commits/f881e2b3fec578fbeeb29dfaf2e8badef7bd8900 (see also https://github.com/sagemath/sage/pull/37737#issuecomment-2102989850).
